samples/ChatbotExamplev2/Services/PersonalizerCoffeeClient.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.CognitiveServices.Personalizer;
+using Microsoft.Azure.CognitiveServices.Personalizer.Models;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ChatbotSample.Services
```

Comment:
I've kept the creation of the PersonalizerClient service as-is in Startup for now, but I've created this class as a possible alternate approach and would like some feedback; I see that in the example I'm working off of, they created a sort of wrapper class for the LuisRecognizer that extended the same interface that LuisRecognizer does and took care of instantiating it using the config values from appsettings (this class is now CoffeeRecognizer in this sample). I'm wondering if we should use the same approach for Personalizer. I like the idea of taking the details of how PersonalizerClient is created out of Startup for a "cleaner" approach; however, stubbing out the IPersonalizerClient involves implementing a lot of methods that this sample will never use (this is why I extended IDispose here instead, although I think extending IPersonalizerClient communicates better what this class is trying to do.) Also, I'm not sure that having a wrapper class that essentially just instantiates another class and calls all that other class's methods without any sort of additional modification is a good style choice. Let me know your thoughts

Reply:
I think what you want here is a factory. What you have here is more like a Factory + Decorator/Adapter that doesn't decorate or adapt anything (that I can tell). I don't know all the interfaces that are at play here, but I'm guessing if you make this a simple factory, the design will fall into a place. In general, having a factory to abstract creation is a "safe" design choice 😄

Reply:
I ultimately decided against having any sort of factory/wrapper; it was starting to feel like overkill. I moved the creation of LuisRecognizer and PersonalizerClient both to anonymous functions inside of addSingleton() instead